oauth-server/src/main/java/io/strimzi/kafka/oauth/server/OAuthSessionAuthorizer.java

```diff
@@ -0,0 +1,159 @@
+ * </pre>
+ * The specified delegate authorizer should be configured according to its documentation, as if it was installed as the
+ * main authorizer (using <em>authorizer.class.name</em>).
+ * <p>
+ * If the `strimzi.authorizer.delegate.class.name` is not set, any operation will be granted as long as the the access token
+ * has not yet expired.
+ * </p>
+ */
+@SuppressWarnings("deprecation")
+public class OAuthSessionAuthorizer implements kafka.security.auth.Authorizer {
+
+    static final Logger log = LoggerFactory.getLogger(OAuthSessionAuthorizer.class);
+    static final Logger GRANT_LOG = LoggerFactory.getLogger(OAuthSessionAuthorizer.class.getName() + ".grant");
+    static final Logger DENY_LOG = LoggerFactory.getLogger(OAuthSessionAuthorizer.class.getName() + ".deny");
+
+    private kafka.security.auth.Authorizer delegate;
+
+    @Override
+    public void configure(java.util.Map<String, ?> configs) {
+        String className = (String) configs.get(ServerConfig.STRIMZI_AUTHORIZER_DELEGATE_CLASS_NAME);
+
+        if (className != null) {
+            try {
+                Class delegateClass = Thread.currentThread().getContextClassLoader().loadClass(className);
+                if (!kafka.security.auth.Authorizer.class.isAssignableFrom(delegateClass)) {
+                    throw new IllegalArgumentException("The class specified by " + ServerConfig.STRIMZI_AUTHORIZER_DELEGATE_CLASS_NAME + " is not an instance of kafka.security.auth.Authorizer");
+                }
+
+                delegate = (kafka.security.auth.Authorizer) delegateClass.<kafka.security.auth.Authorizer>newInstance();
+
+                // Configure the delegate
+                delegate.configure(configs);
+
+            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                throw new RuntimeException("Failed to instantiate and configure the delegate authorizer: " + className, e);
+            }
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Configured OAuthSessionAuthorizer:"
+                    + "\n    " + ServerConfig.STRIMZI_AUTHORIZER_DELEGATE_CLASS_NAME + ": " + className);
+        }
+    }
+
+    @Override
+    public boolean authorize(RequestChannel.Session session, kafka.security.auth.Operation operation, kafka.security.auth.Resource resource) {
+
+        KafkaPrincipal principal = session.principal();
+
+        if (!(principal instanceof OAuthKafkaPrincipal)) {
+            // If user wasn't authenticated over OAuth, there's nothing for us to check
+            if (delegate != null) {
+                return delegate.authorize(session, operation, resource);
+            } else {
+                if (GRANT_LOG.isDebugEnabled()) {
+                    GRANT_LOG.debug("Authorization GRANTED - no access token: " + session.principal() + ", operation: " + operation + ", resource: " + resource);
+                }
+                return true;
```

Comment:
I can see that you want the delegate to be optional, which sort-of makes sense, but I can also see that the user is one typo away from a default allow, which makes me uncomfortable.

Reply:
Good point. We could add another option e.g. strimzi.authorizer.deny.if.no.delegate=false.
If option is not specified this way we would return false when no delegate is configured.

Reply:
Yeah, that would work. Maybe we should also WARN about configs beginning with `strimzi.authorizer` which this class doesn't understand (or are you expecting the delegate might also consume `strimzi.authorizer.*` configs)?

Reply:
I'd say let's not overdo this. Users should read the docs, and the docs should be obvious about options for specific authorizer.